nf_core/components/nfcore_component.py

```diff
@@ -90,3 +94,47 @@ def _get_included_components(self, main_nf: str):
                 if line.strip().startswith("include"):
                     included_components.append(line.strip().split()[-1].split(self.org)[-1].split("main")[0].strip("/"))
         return included_components
+
+    def get_inputs_from_main_nf(self):
+        """Collect all inputs from the main.nf file."""
+        inputs = []
+        with open(self.main_nf, "r") as f:
+            data = f.read()
+        # get input values from main.nf after "input:", which can be formatted as tuple val(foo) path(bar) or val foo or val bar or path bar or path foo
+        # regex matches:
+        # val(foo)
+        # path(bar)
+        # val foo
+        # val bar
+        # path bar
+        # path foo
+        # don't match anything inside comments or after "output:"
+        if "input:" not in data:
+            log.debug(f"Could not find any inputs in {self.main_nf}")
+            return inputs
+        input_data = data.split("input:")[1].split("output:")[0]
+        regex = r"(val|path)\s*(\(([^)]+)\)|\s*([^)\s]+))"
+        matches = re.finditer(regex, input_data, re.MULTILINE)
+        for matchNum, match in enumerate(matches, start=1):
+            if match.group(3):
+                inputs.append(match.group(3))
+            elif match.group(4):
+                inputs.append(match.group(4))
+        log.info(f"Found {len(inputs)} inputs in {self.main_nf}")
+        self.inputs = inputs
+
+    def get_outputs_from_main_nf(self):
+        outputs = []
+        with open(self.main_nf, "r") as f:
+            data = f.read()
+        # get output values from main.nf after "output:". the names are always after "emit:"
+        if "output:" not in data:
+            log.info(f"Could not find any outputs in {self.main_nf}")
+            return outputs
+        output_data = data.split("output:")[1].split("when:")[0]
+        regex = r"emit:\s*([^)\s]+)"
```

Comment:
Like in input, it includes commas after the output name. https://regex101.com/r/1TXYni/1
```suggestion
        regex = r"emit:\s*([^)\s,]+)"
```